### PR TITLE
Winners page tests

### LINKED ISSUE #477: Bug (in develop only) – winners page not loading data
This is a new javascript bug that was likely caused by #465 when new js lint specifications were added 

screen shot 2016-04-27 at 3 35 23 pm

### PR DESCRIPTION
**Pull request by gemfarmer**
For #456 

This PR does some clean up to the front end code, and adds jasmine unit tests to improve future maintenance.

PR adds: 

- [route specs](https://github.com/18F/micropurchase/pull/459/files#diff-2088da31c15f1f3b97ea169cf15cdadeL39)
- [spec for drawer.js](https://github.com/18F/micropurchase/pull/459/files#r61348812)
- [spec for previousWinners.js](https://github.com/18F/micropurchase/pull/459/files#r61348878)
- [spec for utility.js](https://github.com/18F/micropurchase/pull/459/files#r61348963)
- mock data
- refactored previousWinners.js
- refactored utility.js
- Fixes #477 (console error in dev caused by linting)

Ideally we wait until #453 has completed to do a complete refactor of previousWinners to accommodate `winning_bid` logic separately for #483

Ready for review! The following tests need to pass and `/auctions/winners/` and `/auctions/winners/archive/` pages need visual spot checks
- jasmine
- cucumber
- rspec

### CODE REVIEW

**Comment by gemfarmer** [comment]
screen shot 2016-04-22 at 3 01 05 pm

File: spec/javascripts/drawer_spec.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+describe('Drawer', function () {
```
Comment (gemfarmer) [inline spec/javascripts/drawer_spec.js]: spec for drawer.js

File: spec/javascripts/previous_winners_spec.js [hunk 2]
```diff
@@ -0,0 +1,242 @@
+describe('PreviousWinners', function () {
```
Comment (gemfarmer) [inline spec/javascripts/previous_winners_spec.js]: spec for previousWinners.js

File: spec/javascripts/utility_spec.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
+describe('Utility', function () {
```
Comment (gemfarmer) [inline spec/javascripts/utility_spec.js]: spec for utility.js

**Comment by gemfarmer** [comment]
Finally ready for review!

**Comment by gemfarmer** [comment]
@jessieay thanks for the review! Can it be merged?

**Comment by jessieay** [comment]
One change I'd like to see before merging:

many of the cucumber spec expectations are written like this

```ruby
expect(previous_auction_item).to have_content
```

we need to pass something to these expectations, otherwise they will always pass and aren't really testing anything. Should be looking for text or css selectors.

**Comment by gemfarmer** [comment]
@jessieay unless there are instances I'm not seeing, I think every time I am doing that I am passing in a selector. [example](https://github.com/18F/micropurchase/pull/459/files#diff-bc7016cd71432d5abf12a5ecefaba4cfR24)

**Comment by jessieay** [comment]
@gemfarmer ah ok I was not seeing that bc this is how people would usually write it:

```ruby
auctions_total = some_selector
within auctions_total do
  expect(page).to have("this specific content")
end
```

if there is no specific content you are looking for within a selector, you can do:

```ruby
auctions_total = some_selector
expect(page).to have(auctions_total)
```

you mind doing moving stuff around to look like one of the two examples above? I don't think what you have now is wrong but it is not how I usually see capybara expectations written

**Comment by gemfarmer** [comment]
@jessieay well I have never written capybara expectations before, so that would explain your confusion. I'll get to it by the end of the day :)

File: app/assets/javascripts/utility.js [hunk 4]
```diff
@@ -55,14 +55,18 @@
   };
 
   microp.format.removeGitPrefix = function(str) {
-    return str.split('https://github.com/').join('');
+    return str.includes('https')
```
Comment (adelevie) [inline app/assets/javascripts/utility.js]: this is all fine, but maybe we should eventually extract a standalone js library for this stuff.

File: features/previous_winners.feature [hunk 5]
```diff
@@ -0,0 +1,32 @@
+Feature: Previous Winners View and Previous Winners Archive Page
+  As an unauthenticated user
+  I want to be able to previous winners on the site
```
Comment (adelevie) [inline features/previous_winners.feature]: I think this is missing a verb here.

File: features/step_definitions/previous_winners_steps.rb [hunk 6]
```diff
@@ -0,0 +1,100 @@
+When(/^I visit the previous winners page$/) do
+  visit "/auctions/winners/"
+end
+
+When(/^I visit the previous winners archive page$/) do
+  visit "/auctions/winners/archive/"
+end
+
+Then(/^I should visit the previous winners archive page$/) do
```
Comment (adelevie) [inline features/step_definitions/previous_winners_steps.rb]: I **think** this is wrong. "Should" implies an assertion to be tested. I think you might be using this to test a redirect?

File: spec/javascripts/previous_winners_spec.js (hunk 2 above)
Comment (adelevie) [inline spec/javascripts/previous_winners_spec.js]: I think the spacing is off, or at least inconsistent. We should probably use two-space indents.

File: spec/javascripts/drawer_spec.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    });
+	});
+
+	describe('When show is triggered', function () {
+	  beforeEach(function () {
+	    drawer = new Drawer();
+	    drawer.show()
+	  });
+
+	  it('then the drawer is open', function() {
```
Comment (adelevie) [inline spec/javascripts/drawer_spec.js]: I think the spacing here needs to be fixed.

File: app/assets/javascripts/utility.js (hunk 4 above)
Comment (gemfarmer) [inline app/assets/javascripts/utility.js]: @adelevie for sure. @vzvenyach recently introduced me to https://github.com/18F/micropurchasedata. There isn't much too many methods there right now, but a lot of these methods could be moved there. I created #480 for this is

File: spec/javascripts/drawer_spec.js (hunk 7 above)
Comment (gemfarmer) [inline spec/javascripts/drawer_spec.js]: ah! mixed spaces/tabbing issue. are we a spaces or tabs repo?

Comment (jessieay) [inline spec/javascripts/drawer_spec.js]: spaces please!

File: features/previous_winners.feature (hunk 5 above)
Comment (gemfarmer) [inline features/previous_winners.feature]: 👍

File: features/step_definitions/previous_winners_steps.rb (hunk 6 above)
Comment (gemfarmer) [inline features/step_definitions/previous_winners_steps.rb]: This sounds right.

**Comment by gemfarmer** [comment]
@jessieay @adelevie changes to cucumber made! Should be ready to merge. The tests should pass once #490 is merged

**Comment by jessieay** [comment]
MERGING

**Comment by gemfarmer** [comment]
Thanks @jessieay !

**Pull request merged**